tests/key_interrupt/test_key_interrupt.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright 2024 Harrison Chan (@xelus22)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "keycode.h"
+#include "test_common.hpp"
+
+using ::testing::_;
+using ::testing::AnyNumber;
+using ::testing::InSequence;
+
+class KeyInterrupt : public TestFixture {
+   public:
+    void SetUp() override {
+        key_interrupt_enable();
+    }
+    // Convenience function to tap `key`.
+    void TapKey(KeymapKey key) {
+        key.press();
+        run_one_scan_loop();
+        key.release();
+        run_one_scan_loop();
+    }
+
+    // Taps in order each key in `keys`.
+    template <typename... Ts>
+    void TapKeys(Ts... keys) {
+        for (KeymapKey key : {keys...}) {
+            TapKey(key);
+        }
+    }
+};
+
+// Test that verifies enable/disable/toggling works
+TEST_F(KeyInterrupt, OnOffToggle) {
+    TestDriver driver;
```

Comment:
This test doesn't send any hid reports, please remove the `TestDriver` and any interactions with it.